### ISSUE TITLE
Ember-cli livereload

### ISSUE THREAD

**Issue by kevinansfield** [TryGhost/Ghost#8161]
### Issue Summary

As of these changes (https://github.com/TryGhost/Ghost/pull/8142 + https://github.com/TryGhost/Ghost-Admin/pull/574) it's now possible to run `ember serve` because the client build pipeline is completely independent from Ghost itself.

`ember serve` has built-in livereload functionality that would be very useful to speed up client development and styling but there are a few issues to work through first:

- `ember serve` starts it's own server on http://localhost:4200, this allows it to boot up and inject the necessary livereload server and script
- it's possible to proxy api requests to the ghost server with `ember serve --proxy=http://localhost:2368` which works fine except that we now have a mismatch between the url being used and the url configured in Ghost which presents some issues around auth (especially when using OAuth)
- alternatively, `ember serve` will still generate output in a way that can be consumed via the normal http://localhost:2368/ghost/ route but then livereload won't work because the script injections are relative

This needs some investigation so that we can update the tooling or otherwise document a way to run Ghost + Ghost-Admin with livereload.

- [ ] is there a config where `ember serve --proxy=http://localhost:2368` will work when accessing http://localhost:4200?
- [x] is there a way to instruct `ember-cli-inject-live-reload` to provide absolute paths for the livereload script and server?
  - Yes and no, it's possible to pass `--live-reload-base-url=x` to `ember-serve` but `ember-cli-inject-live-reload` will fail if that is an absolute URL because it needs to set up a relative URL to be able to respond to the `/:baseurl/ember-cli-live-reload.js` request
- [ ] if neither of the above are possible, is there a simple way to have Ghost proxy/redirect requests when running `grunt dev` so that the default http://localhost:2368/ghost/ path works with livereload?

**Comment by acburdine** [comment]
Ember-CLI has this blueprint: https://github.com/ember-cli/ember-cli/tree/master/blueprints/http-proxy -> which might suit our use-cases

**Comment by kevinansfield** [comment]
I've managed to hack together a solution by adding this to `core/server/admin/app.js` after the service-worker route:

```
// Ember CLI's live-reload script
    adminApp.get('/ember-cli-live-reload.js', function(req, res) {
        res.redirect('http://localhost:4200/ghost/ember-cli-live-reload.js');
    });
``
Then launching `ember serve` with `ember serve --live-reload-base-url=/ghost/ --live-reload-port=4201` (not sure the port is necessary, I was just getting conflict/permission errors with the default port).

However this doesn't seem to be working as well as I expected, CSS changes are forcing a full page refresh rather than being live reloaded - if we can't get around that then we may as well use a live reload via `grunt dev` and watch the `default.hbs` and `core/built/assets/*` files.

**Comment by kevinansfield** [comment]
I've been using the above setup today and it's working well so far (minus the full page refreshes for CSS). One thing that is really useful is the ability to run the tests without having to shut down ghost and then do a rebuild when you want to stop the tests and switch back.

http://localhost:2368/ghost/ = normal blog
http://localhost:4200/tests = client tests with livereload (you can still filter too, e.g. http://localhost:4200/tests?grep=gh-publishmenu)

**Comment by kevinansfield** [comment]
I've been hitting a rather annoying bug that is only affecting touchbar macs 🙄 https://github.com/ember-cli/ember-cli/issues/6513

Until that's resolved (it sounds like there are a couple of promising paths) we may have to set a fixed livereload port

**Comment by kevinansfield** [comment]
Closing as this has now been implemented via #8176 and TryGhost/Ghost-Admin#590
